### PR TITLE
Create new project #25

### LINKED ISSUE #25: Create project views and tests.
An authenticated user can open new project: project title, date are needed. List of linked participants, list of linked tasks are possible. Opening new project automatically gives user's project manager privileges, makes him also a participant and an author.

### PR DESCRIPTION
**Pull request by karolcircle**
closes #25

### CODE REVIEW

File: app/views/projects/index.html.erb [hunk 1]
```diff
@@ -1,2 +1,27 @@
...
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @projects.each do |project| %>
+      <tr>
+        <td><%= project.title %></td>
+        <td><%= link_to 'Show', project %></td>
+        <td><%= link_to 'Edit', edit_project_path(project) %></td>
+        <td><%= link_to 'Destroy', project, method: :delete, data: { confirm: 'Are you sure?' } %></td>
```
Comment (filip373) [inline app/views/projects/index.html.erb]: I think that above links should only be visible to: project member (show) and project manager (edit, destroy).

File: app/views/projects/show.html.erb [hunk 2]
```diff
@@ -1,2 +1,17 @@
...
+  Description:
+  <%= @project.description %>
+</p>
+<p>
+  Date:
+  <%= @project.date.strftime("%d/%m/%Y") %>
+</p>
+
+<%= link_to 'Edit', edit_project_path(@project) %> |
+<%= link_to 'Back', projects_path %>
```
Comment (tprochniewicz) [inline app/views/projects/show.html.erb]: add link to tasks index

Comment (tprochniewicz) [inline app/views/projects/show.html.erb]: and events index

File: app/views/projects/edit.html.erb [hunk 3]
```diff
@@ -1,2 +1,5 @@
-<h1>Projects#edit</h1>
-<p>Find me in app/views/projects/edit.html.erb</p>
+<h1>Edit project</h1>
+<%= render "form",  project: @project %>
+
+<%= link_to 'Show', @project %> |
```
Comment (tprochniewicz) [inline app/views/projects/edit.html.erb]: we use  " "

Comment (karolcircle) [inline app/views/projects/edit.html.erb]: ok, I will check also in other files and fix

File: app/views/projects/index.html.erb (hunk 1 above)
Comment (filip373) [inline app/views/projects/index.html.erb]: It's ok, I will handle this authorization policy while doing task with Pundit authorization cleanup.

File: app/views/projects/show.html.erb [hunk 4]
```diff
@@ -1,2 +1,19 @@
-<h1>Projects#show</h1>
-<p>Find me in app/views/projects/show.html.erb</p>
+<p id="notice"><%= notice %></p>
```
Comment (pospieszynski) [inline app/views/projects/show.html.erb]: remove notice , you have acces to it in every view due to the fact that it inherits form application.html.erb

File: app/views/projects/index.html.erb [hunk 5]
```diff
@@ -1,2 +1,27 @@
-<h1>Projects#index</h1>
-<p>Find me in app/views/projects/index.html.erb</p>
+<p id="notice"><%= notice %></p>
```
Comment (tprochniewicz) [inline app/views/projects/index.html.erb]: same as in show

File: app/views/projects/index.html.erb (hunk 1 above)
Comment (karolcircle) [inline app/views/projects/index.html.erb]: ok

File: spec/rails_helper.rb [hunk 6]
```diff
@@ -30,6 +30,7 @@
 
 RSpec.configure do |config|
   config.include Devise::Test::ControllerHelpers, type: :controller
+  config.include Devise::TestHelpers, :type => :view
```
Comment (tprochniewicz) [inline spec/rails_helper.rb]: change it to:
config.include Devise::Test::ControllerHelpers, type: :view

**Comment by pospieszynski** [comment]
:+1: 
@gotar

File: app/views/projects/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,28 @@
+<%= form_for(project) do |f| %>
+  <% if project.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(project.errors.count, "error") %> prohibited this project from being saved:</h2>
+      <ul>
+        <% project.errors.full_messages.each do |message| %>
+        <li><%= message %></li>
```
Comment (gotar) [inline app/views/projects/_form.html.erb]: wrong indentation

**Comment by gotar** [comment]
squash and you can merge

**Pull request merged**